postprocessing/comstockpostproc/utils/hpc.py

```diff
@@ -654,43 +655,121 @@ def count_job_errs(job_err_path, errs_dir):
+                        if data.get('timeout'):
+                            error_lines.append(f'\tTimeout: {data.get("timeout")}\n')
+                        steps = data.get('steps', [])
+                        for step in steps:
+                            result = step.get('result', {})
+                            measure_name = result.get('measure_name')
+                            step_errors = result.get('step_errors', [])
+                            if step_errors:
+                                error_lines.append(f'\tIn step: {measure_name}:\n')
+                                for step_error in step_errors:
+                                    if '[openstudio.standards.command]' in step_error:
+                                        # these aren't that useful
+                                        continue
+                                    if 'Sizing run for Hardsize model failed, cannot hard-size model.' in step_error:
+                                        error_lines.append(f'\t{step_error}\n')
+                                        # actual error message in stdout
+                                        std_out = result.get('stdout')
+                                        lines = std_out.split('\n')
+                                        for line in lines:
+                                            if '[openstudio.model.Model]' in line:
+                                                line = re.sub(r'(\\n)|(\n)',r'\n\t\t', line)
+                                                error_lines.append(f'\t\t{line}\n')
+                                    else:
+                                        # indent and newline the messages for readability
+                                        step_error = re.sub(r'(\\n)|(\n)',r'\n\t\t', step_error)
+                                        error_lines.append(f'\t\t{step_error}\n')
+                except KeyError:
+                    try:
+                        sing_out_bytes=tar.extractfile(f'up{job[0]}/bldg{job[1]}/openstudio_output.log').read()
+                        for l in sing_out_bytes.decode().split('\n'):
+                            if re.search(r'\[.* ERROR\]|\[.*<Error>.*\]', l):
+                                error_lines.append(f'\t\t{l}\n')
+                    except KeyError as err:
+                        error_lines.append(f'Did not find out.osw or openstudio_output.log for up{job[0]}/bldg{job[1]}, cannot extract failure details\n')
+                if f'up{job[0]}' not in errors_dict:
+                    errors_dict[f'up{job[0]}'] = {}
+                    if f'bldg{job[1]}' not in errors_dict[f'up{job[0]}']:
+                        errors_dict[f'up{job[0]}'][f'bldg{job[1]}'] = error_lines
+                        all_errors = merge_error_dicts(all_errors, errors_dict)
+            write_sorted_dict(all_errors, sort_order, job_fails_path)
+            return all_errors
+        tar.close()
```

Comment:
tar.close() should in the scope `with tarfile.open(tar_path) as tar:`.

Reply:
Or actually the close is not nessessary. https://stackoverflow.com/questions/21275836/if-youre-opening-a-file-using-the-with-statement-do-you-still-need-to-close 

`with` keyword save the cope